adapters/triplelift_native/triplelift_native.go

```diff
@@ -0,0 +1,195 @@
+package triplelift_native
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type TripleliftNativeAdapter struct {
+	endpoint string
+	extInfo  TripleliftNativeExtInfo
+}
+
+type TripleliftInnerExt struct {
+	Format int `json:"format"`
+}
+
+type TripleliftRespExt struct {
+	Triplelift TripleliftInnerExt `json:"triplelift_pb"`
+}
+
+type TripleliftNativeExtInfo struct {
+	// Array is used for deserialization.
+	PublisherWhitelist []string `json:"publisher_whitelist"`
+
+	Endpoint string `json:"endpoint"`
+
+	// Map is used for optimized memory access and should be constructed after deserialization.
+	PublisherWhitelistMap map[string]bool
+}
+
+func getBidType(ext TripleliftRespExt) openrtb_ext.BidType {
+	return openrtb_ext.BidTypeNative
+}
+
+func processImp(imp *openrtb.Imp) error {
+	// get the triplelift extension
+	var ext adapters.ExtImpBidder
+	var tlext openrtb_ext.ExtImpTriplelift
+	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
+		return err
+	}
+	if err := json.Unmarshal(ext.Bidder, &tlext); err != nil {
+		return err
+	}
+	if imp.Native == nil {
+		return fmt.Errorf("no native object specified")
+	}
+	imp.TagID = tlext.InvCode
```

Comment:
Since `InvCode` is a required parameter, it will be good to check here if it is empty or not and if it is, you can just return an error here and fail fast rather than this reaching the TripleLift API and then failing.

Reply:
makes sense--done